app-backend/datamodel/src/main/scala/com/azavea/rf/datamodel/package.scala

```diff
@@ -18,6 +18,19 @@ import geotrellis.proj4._
 import geotrellis.slick.Projected
 
 package object datamodel {
+
+  trait OwnerCheck {
+    def checkOwner(createUser: User, ownerUserId: Option[String]): String = {
+      (createUser, ownerUserId) match {
+        case (user, Some(id)) if user.id == id => user.id
+        case (user, Some(id)) if user.isInRootOrganization => id
+        case (user, Some(id)) if !user.isInRootOrganization =>
+          throw new IllegalArgumentException("Insufficient permissions to set owner on object")
```

Comment:
That's a good idea -- I'm going to handle this in a separate PR and created https://github.com/azavea/raster-foundry/issues/1591 to make sure we don't forget. @alkamin is working on something that needs these changes so I want to get it merged now